cpp_linter/clang_tools/clang_format.py

```
@@ -200,6 +203,23 @@ def run_clang_format(
     advice = parse_format_replacements_xml(
         results.stdout.decode(encoding="utf-8").strip(), file_obj, lines_changed_only
     )
+    if fix and advice.replaced_lines:
+        # Reuse the assembled (range-aware) args to apply fixes in-place. This
+        # runs in the same parallel worker as the analysis above, and honors
+        # --lines-changed-only via the --lines args already in `cmds`.
+        fix_cmds = [arg for arg in cmds if arg != "--output-replacements-xml"]
+        fix_cmds.insert(-1, "-i")  # apply edits in-place, just before the file path
+        logger.info('Applying formatting fixes with "%s"', " ".join(fix_cmds))
+        fix_results = subprocess.run(fix_cmds, capture_output=True)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)